# Connect to Google Drive

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import os
myDirFiles = "/content/drive/MyDrive/Colab Notebooks/Digital Skola/6.Statistics"
os.chdir(myDirFiles)

# Import Libraries

In [3]:
import pandas as pd
import os
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as st
import scipy

# Case 1. House Prices Advanced Regression

In [7]:
# Load the data
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Digital Skola/6.Statistics/house-prices-advanced-regression-techniques.zip (Unzipped Files)/train.csv')
df.head(5)

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [8]:
ndf = df.fillna(0)

In [9]:
dfFilter = ndf.groupby('YrSold')['SalePrice'].mean().to_frame('SalePrice').reset_index()
dfFilter

YrSold      SalePrice
0    2006  182549.458599
1    2007  186063.151976
2    2008  177360.838816
3    2009  179432.103550
4    2010  177393.674286

Apakah YrSold berpengaruh terhadap harga jual dan Setiap tahunnya harga rumah mengalami kenaikan?

# Penentuan Hipotesis

Uji Hipotesis sisi kiri

H0 : harga rumah <= harga tahun sebelumnya

H1 : harga rumah > harga tahun sebelumnya

α = 5% = 0.05

In [10]:
y2006 = ndf[ndf['YrSold']==2006]
y2007 = ndf[ndf['YrSold']==2007]
y2008 = ndf[ndf['YrSold']==2008]
y2009 = ndf[ndf['YrSold']==2009]
y2010 = ndf[ndf['YrSold']==2010]

In [11]:
anova_test = st.f_oneway(y2006['SalePrice'],y2007['SalePrice'],y2008['SalePrice'],y2009['SalePrice'],y2010['SalePrice'])

In [12]:
anova_test.pvalue

0.6300887663573457

In [13]:
if anova_test.pvalue > 0.05:
    print('Ho diterma')
else:
    print('Ho ditolak')

Ho diterma


Kesimpulan : Ho diterima, sehingga harga rumah tidak selalu naik setiap tahunnya.

# Case 2. Open a deposit account for bank customers

In [16]:
# Load the data
dfBank = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Digital Skola/6.Statistics/bank-full.csv', sep=';')
dfBank.head(5)

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

In [17]:
dfBank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        45211 non-null  int64 
 1   job        45211 non-null  object
 2   marital    45211 non-null  object
 3   education  45211 non-null  object
 4   default    45211 non-null  object
 5   balance    45211 non-null  int64 
 6   housing    45211 non-null  object
 7   loan       45211 non-null  object
 8   contact    45211 non-null  object
 9   day        45211 non-null  int64 
 10  month      45211 non-null  object
 11  duration   45211 non-null  int64 
 12  campaign   45211 non-null  int64 
 13  pdays      45211 non-null  int64 
 14  previous   45211 non-null  int64 
 15  poutcome   45211 non-null  object
 16  y          45211 non-null  object
dtypes: int64(7), object(10)
memory usage: 5.9+ MB


In [18]:
dfBank.describe()

age        balance           day      duration      campaign  \
count  45211.000000   45211.000000  45211.000000  45211.000000  45211.000000   
mean      40.936210    1362.272058     15.806419    258.163080      2.763841   
std       10.618762    3044.765829      8.322476    257.527812      3.098021   
min       18.000000   -8019.000000      1.000000      0.000000      1.000000   
25%       33.000000      72.000000      8.000000    103.000000      1.000000   
50%       39.000000     448.000000     16.000000    180.000000      2.000000   
75%       48.000000    1428.000000     21.000000    319.000000      3.000000   
max       95.000000  102127.000000     31.000000   4918.000000     63.000000   

              pdays      previous  
count  45211.000000  45211.000000  
mean      40.197828      0.580323  
std      100.128746      2.303441  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       -1.000000      0.000000  
max      871.000000    275.000000

In [22]:
dfBankFilter = dfBank[['campaign', 'y']]
dfBankFilter.fillna(0)
dfBankFilter

campaign    y
0             1   no
1             1   no
2             1   no
3             1   no
4             1   no
...         ...  ...
45206         3  yes
45207         2  yes
45208         5  yes
45209         4   no
45210         2   no

[45211 rows x 2 columns]

Apakah jumlah Campaign mampu meningkatkan keinginan user untuk membuka akun deposit atau tidak?

a. Apa nama experimentnya? 

Menentukan hubungan antara jumlah campaign dengan keinginan user untuk deposit    

b. Definisikan Hipotesis.

Null_hypothesis = "Tidak ada hubungan antara jumlah campaign dengan keinginan user untuk deposit"
Alternate_hypothesis = "Ada hubungan antara jumlah campaign dengan keinginan user untuk deposit"

c. Siapa participant-nya? 

user bank

d. Variabel yang akan diuji apa?

Campaign

e. Metrics apa yang akan digunakan? 

Jumlah campaign

f. Berapa sample size dan durasi experiment? 

45211,  0.030620574951171875 detik
